File: DataBindingStudy/app/src/main/java/com/jie/databinding/ClassMethodActivity.java
package com.jie.databinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.text.Editable;
import android.widget.Toast;

import com.jie.databinding.databinding.ActivityClassMethodBinding;
import com.jie.databinding.databinding.ActivityEventBindingBinding;
import com.jie.databinding.model.User;

public class ClassMethodActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityClassMethodBinding methodBinding = DataBindingUtil.setContentView(this, R.layout.activity_class_method);
        User userinfo = new User("FrankieAngelica", "123654");
        UserPresenter userPresenter = new UserPresenter();
        methodBinding.setUserInfo(userinfo);
        methodBinding.setUserPresenter(userPresenter);
        setTitle("class method");
    }

    public class UserPresenter {
        public void onUserNameClick(User user) {
            Toast.makeText(ClassMethodActivity.this, "用户名:" + user.getName(), Toast.LENGTH_LONG).show();
        }

    }
}

File: DataBindingStudy/app/src/main/java/com/jie/databinding/ui/activity/ClassMethodActivity.kt
package com.jie.databinding.ui.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import com.jie.databinding.R
import com.jie.databinding.databinding.ActivityClassMethodBinding
import com.jie.databinding.model.UserEntity

class ClassMethodActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DataBindingUtil.setContentView<ActivityClassMethodBinding>(
            this,
            R.layout.activity_class_method
        ).apply {
            val userInfo: UserEntity = UserEntity("FrankieAngelica", "123654")
            this.userInfo = userInfo
            val userPresenter = UserPresenter(this@ClassMethodActivity)
            this.userPresenter = userPresenter
        }
        title = "class method study"
    }

    inner class UserPresenter constructor(val context: Context) {
        fun onUserNameClick(userInfo: UserEntity) {
            Toast.makeText(context, "用户名:${userInfo.name}", Toast.LENGTH_SHORT).show()
        }
    }
}